ListView-HW/app/src/main/java/com/joelimyx/listview_hw/MainActivity.java

```diff
@@ -0,0 +1,131 @@
+        spinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinner.setAdapter(spinnerAdapter);
+
+        //ListView Adapter
+        myAdapter = new BaseAdapter() {
+            @Override
+            public int getCount() {
+                return mPlayerList.size();
+            }
+
+            @Override
+            public Object getItem(int position) {
+                return mPlayerList.get(position);
+            }
+
+            @Override
+            public long getItemId(int position) {
+                return position;
+            }
+
+            @Override
+            public View getView(int position, View convertView, ViewGroup parent) {
+                if(convertView==null){
+                    LayoutInflater inflater = LayoutInflater.from(MainActivity.this);
+                    convertView = inflater.inflate(R.layout.player_adapter,null);
+                }
+                TextView textView = (TextView) convertView.findViewById(R.id.text1);
+                textView.setText(mPlayerList.get(position).getName());
+                TextView textView2 = (TextView) convertView.findViewById(R.id.text2);
+                textView2.setText(mPlayerList.get(position).getRole());
+                return convertView;
+            }
+        };
+
+        // spinner Onclick
+        spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                mRole = (String) parent.getSelectedItem();
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+
+            }
+        });
+
+        mListView.setAdapter(myAdapter);
+        mAddButton.setOnClickListener(this);
+        mRemoveButton.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()){
+            //Add Player
+            case R.id.addButton:
+                String temp = mEditText.getText().toString();
+                boolean isRepeat=false;
+                //Check Empty
+                if (temp.isEmpty()){
+                    mEditText.setError("Please Enter a Name");
+                }else {
+                    //Loop thru and check if repetition
+                    for (Player player : mPlayerList) {
+                        if (player.getName().equalsIgnoreCase(temp)&&player.getRole().equalsIgnoreCase(mRole)){
+                            mEditText.setError("Do not enter the same person and role.");
+                            isRepeat = true;
+                        }
+                    }
+                }
+
+                if(!(isRepeat||temp.isEmpty())){
+                    mPlayerList.add(new Player(temp, mRole));
```

Comment:
Here, instead of `mRole` you can call `spinner.getSelectedItem().toString()` - but your way works too :)